### ISSUE TITLE
slave.jar problem

### ISSUE THREAD

**Issue by adrianlop** [iverberk/jenkins-nomad#2]
Hi @iverberk,
I'm having a problem launching jenkins slaves in Nomad:
Error: Unable to access jarfile /local/local/slave.jar

I'm using a custom image with java+scala+sbt, but it seems it can't download the jar.
I also tried copying the slave.jar inside this image, inside /local/local/slave.jar, but didn't work either.

Any clue? thanks!

**Comment by iverberk** [comment]
Hi @adrianlop!

I'll look into it asap. Which version of Nomad are you running? And can you share the Dockerfile with me? That'll help me debug this issue for you.

**Comment by adrianlop** [comment]
I'm using Nomad v0.3.0, and the docker image is `hseeberger/scala-sbt`. I also tried adding `slave.jar` with this custom Dockerfile, but no luck:
```
FROM hseeberger/scala-sbt

RUN mkdir -p /local/local
COPY slave.jar /local/local/slave.jar
RUN chmod -R 777 /local
```
thanks @iverberk !!!

**Comment by iverberk** [comment]
Ah ok. So, the Nomad plugin will not work with Nomad version < 0.3.2 (because this change is needed: https://github.com/hashicorp/nomad/pull/944/files). I compiled from master to make it work in the tutorial (https://github.com/iverberk/nomad-ci/tree/master/nomad/bin) but the change is in 0.3.2 (the latest released version). 

You should not have to do anything special with the Docker image as the Nomad plugin will inject the slave.jar dynamically and start it. Just make sure that you don't use an Entrypoint. So can you try again after upgrading the Nomad version?

**Comment by adrianlop** [comment]
oook that makes sense, thanks! I'll try again.

**Comment by adrianlop** [comment]
@iverberk ok I'm doing some progress. Now with Nomad 0.3.2 the slaves receive the `slave.jar`! but my Jenkins is secured with users/passwords, and I get the following error from the slaves:
```
 Failing to obtain http://10.34.3.175:1234/computer/jenkins-635b4e578fa3/slave-agent.jnlp
 java.io.IOException: Failed to load http://10.34.3.175:1234/computer/jenkins-635b4e578fa3/slave-agent.jnlp: 403 Forbidden
 	at hudson.remoting.Launcher.parseJnlpArguments(Launcher.java:275)
 	at hudson.remoting.Launcher.run(Launcher.java:219)
 	at hudson.remoting.Launcher.main(Launcher.java:192)
 Waiting 10 seconds before retry
```
can I config user/pass in the plugin?
thanks for the help :)

**Comment by adrianlop** [comment]
btw, it works like a charm if I disable Jenkins security :) I could compile a Scala project.
I'll continue testing it.

one more question, can you describe the internals of scaling slaves? I thought that when the build finishes, the slave would die, but it didn't. will the slaves die after some period of inactivity?

great plugin @iverberk

**Comment by iverberk** [comment]
Thanks @adrianlop!

The security part should be added as soon as possible. I didn't need it for my proof of concept but obviously it is necessary for any production usage. I'll try to incorporate it over the next couple of days. Any help is greatly appreciated if you're able to contribute! :-)

The slaves should disappear after a certain amount of idle time. This is configurable on the slave template. Please let me know if this doesn't work for you.

**Comment by adrianlop** [comment]
hi again @iverberk ! do you have an estimated date for auth implementation?
thanks in advance :)

**Comment by pawbur** [comment]
Hi,

I just created PR for using this with Jenkins security enabled https://github.com/jenkinsci/nomad-plugin/pull/1. I'm not a developer but seems to be working fine for me. Don't need any additional configuration, it uses automatically generated slave secrets.

**Comment by iverberk** [comment]
Hi @pawbur,

Thanks for this. I'll review as soon as possible. I have two days of HashiConf EU ahead so it might take a little longer. Very much appreciated!

Regards,
Ivo

**Comment by iverberk** [comment]
@adrianlop I just merged the auth code from @pawbur . I'll look at your other issue and create a new version of the plugin.

**Comment by iverberk** [comment]
This is just released in 0.2.
